### ISSUE TITLE
Add support for conditional module dependencies in jetty-start

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#4628]
**Description**

The jetty-start module mechanism could benefit from having `[depends]` on modules that may or may-not actually be present.

This would simplify behaviors like the ones identified (by @sbordet) with alpn and Java 8 update 252 and newer having built-in support for ALPN (no longer any need for alpn-boot).

Proposed syntax:

```
[depends]
?alpn-boot/alpn-boot-${java.version}
```

That would mean if the `alpn-boot/alpn-boot-${java.version}` module file doesn't exist, then it's a quietly skipped.
(This skipped module is still logged using StartLog's DEBUG level)

This syntax uses the `"?"` character to indicate an optional behavior similar to how it's used with the property syntax `<name>?=<value>` today.

However, the character `"?"` is a valid character on the filesystem. (Does this matter to us?)

**Comment by sbordet** [comment]
See #4443 for ALPN support in 8u252 without `alpn-boot`.

**Comment by joakime** [comment]
would it make more sense to have this be a different section?

```
[speculative]
alpn-boot/alpn-boot-${java.version}
```

I don't know a good name for this section as `[optional]` is already taken.

Some alternate thoughts on names ...

* assume
* rely
* expect
* attach
* entrust
* confer
* delegate
* furnish
* pivot

**Comment by joakime** [comment]
Merged PR #4629

**Comment by gregw** [comment]
@sbordet  How is this different from the existing `[optional]` mechanism?

**Comment by gregw** [comment]
@sbordet has explained the difference to [optional].  An optional dependency is just ordered but not enabled - it has to be explicitly enabled by something else.
This mechanism is for real dependencies that might not exist, but if they do exist they are definitely enabled and ordered.

Thus my issue with this is the name - `NotRequired` is the wrong name as it implies optional.  I think `Conditional` better captures the meaning, as it is a real dependency that is conditional.  In this case the only condition we have syntax for is existence, but in future we might add some `?{otherCondition}dependency` syntax to add other conditions.   

I think the naming can be fixed up in #4633

**Comment by sbordet** [comment]
Fixed with #4633.
